sdk_extensions/zpages/src/main/java/io/opentelemetry/sdk/extensions/zpages/TracezZPageHandler.java

```diff
@@ -0,0 +1,603 @@
+      return;
+    }
+    // Link to OpenTelemetry Logo
+    out.write(
+        "<img style=\"height: 90px;\""
+            + "src=\"data:image/png;base64,"
+            + ZPageLogo.logoBase64
+            + "\" />");
+    out.write("<h1>TraceZ Summary</h1>");
+    Formatter formatter = new Formatter(out, Locale.US);
+    emitSummaryTable(out, formatter);
+    // spanName will be null if the query parameter doesn't exist in the URL
+    String spanName = queryMap.get(PARAM_SPAN_NAME);
+    if (spanName != null) {
+      // Show detailed information for the corresponding span
+      String typeStr = queryMap.get(PARAM_SAMPLE_TYPE);
+      if (typeStr != null) {
+        List<SpanData> spans = null;
+        SampleType type = SampleType.fromString(typeStr);
+        if (type == SampleType.UNKNOWN) {
+          // Type of sample is garbage value
+          return;
+        } else if (type == SampleType.RUNNING) {
+          // Display running span
+          spans = dataAggregator.getRunningSpans(spanName);
+          Collections.sort(spans, new SpanDataComparator(/* incremental= */ true));
+        } else {
+          String subtypeStr = queryMap.get(PARAM_SAMPLE_SUB_TYPE);
+          if (subtypeStr != null) {
+            int subtype = Integer.parseInt(subtypeStr);
+            if (type == SampleType.LATENCY) {
+              if (subtype < 0 || subtype >= LatencyBoundaries.values().length) {
+                // N/A or out-of-bound check for latency based subtype, valid values: [0, 8]
+                return;
+              }
+              // Display latency based span
+              LatencyBoundaries latencyBoundary = LatencyBoundaries.values()[subtype];
+              spans =
+                  dataAggregator.getOkSpans(
+                      spanName,
+                      latencyBoundary.getLatencyLowerBound(),
+                      latencyBoundary.getLatencyUpperBound());
+              Collections.sort(spans, new SpanDataComparator(/* incremental= */ false));
+            } else {
+              if (subtype < 0 || subtype >= CanonicalCode.values().length) {
+                // N/A or out-of-bound cueck for error based subtype, valid values: [0, 15]
+                return;
+              }
+              // Display error based span
+              spans = dataAggregator.getErrorSpans(spanName);
+              Collections.sort(spans, new SpanDataComparator(/* incremental= */ false));
+            }
+          }
+        }
+        out.write("<h2>Span Details</h2>");
+        emitSpanNameAndCount(formatter, spanName, spans == null ? 0 : spans.size(), type);
+
+        if (spans != null) {
+          emitSpanDetails(out, formatter, spans);
+        }
+      }
+    }
+  }
+
+  @Override
+  public void emitHtml(Map<String, String> queryMap, OutputStream outputStream) {
+    // PrintWriter for emiting HTML contents
+    PrintWriter out =
+        new PrintWriter(new BufferedWriter(new OutputStreamWriter(outputStream, Charsets.UTF_8)));
```

Comment:
Think we can just use `PrintStream` since we're going to an `OutputStream` anyways.

Also I'd leave the `BufferedWriter` or `BufferedOutputStream` out here and let callers pass it in, they may not want the buffering.